src/modules/jackrack/plugin_mgr.c

```diff
@@ -617,4 +695,393 @@ lv2_plugin_desc_t *lv2_mgr_get_any_desc(lv2_mgr_t *plugin_mgr, char *id)
+	vst2_index++;
+	//continue;
+      }
+
+      /* check it doesn't already exist */
+      exists = FALSE;
+      for (list = vst2_mgr->all_plugins; list; list = g_slist_next(list)) {
+	other_desc = (vst2_plugin_desc_t *) list->data;
+       
+	if (other_desc->id == effect->uniqueID) {
+	  exists = TRUE;
+	  break;
+	}
+      }
+       
+      if (exists) {
+	mlt_log_info(NULL,
+		     "Plugin %d exists in both '%s' and '%s'; using version in '%s'\n",
+		     effect->uniqueID,
+		     other_desc->object_file,
+		     filename,
+		     other_desc->object_file);
+	vst2_index++;
+	//continue;
+      }
+
+      desc = vst2_plugin_desc_new_with_descriptor(filename, vst2_index, effect);
+      vst2_mgr->all_plugins = g_slist_append(vst2_mgr->all_plugins, desc);
+      vst2_index++;
+      vst2_mgr->plugin_count++;
+     
+      /* print in the splash screen */
+      /* mlt_log_verbose( NULL, "Loaded plugin '%s'\n", desc->name); */
+    }
+
+    /* WIP temporarily disabled */
+    /* err = dlclose(dl_handle); */
+    /* if (err) {
+              mlt_log_warning(NULL,
+                              "%s: error closing object file '%s': %s\n",
+                              __FUNCTION__,
+                              filename,
+                              dlerror());
+          } */
+}
+
+
+static gint vst2_mgr_sort(gconstpointer a, gconstpointer b)
+{
+    const vst2_plugin_desc_t *da;
+    const vst2_plugin_desc_t *db;
+    da = (const vst2_plugin_desc_t *) a;
+    db = (const vst2_plugin_desc_t *) b;
+
+    return strcasecmp(da->name, db->name);
+}
+
+static void vst2_mgr_get_dir_plugins(vst2_mgr_t *vst2_mgr, const char *dir)
+{
+    DIR *dir_stream;
+    struct dirent *dir_entry;
+    char *file_name;
+    int err;
+    size_t dirlen;
+
+    dir_stream = opendir(dir);
+    if (!dir_stream) {
+        /*      mlt_log_warning( NULL, "%s: error opening directory '%s': %s\n",
+               __FUNCTION__, dir, strerror (errno)); */
+        return;
+    }
+
+    dirlen = strlen(dir);
+
+    while ((dir_entry = readdir(dir_stream))) {
+        struct stat info;
+
+        if (strcmp(dir_entry->d_name, ".") == 0
+            || mlt_properties_get(vst2_mgr->blacklist, dir_entry->d_name)
+            || strcmp(dir_entry->d_name, "..") == 0)
+            continue;
+
+        file_name = g_malloc(dirlen + 1 + strlen(dir_entry->d_name) + 1);
+
+        strcpy(file_name, dir);
+        if (file_name[dirlen - 1] == '/')
+            strcpy(file_name + dirlen, dir_entry->d_name);
+        else {
+            file_name[dirlen] = '/';
+            strcpy(file_name + dirlen + 1, dir_entry->d_name);
+        }
+
+        stat(file_name, &info);
+        if (S_ISDIR(info.st_mode))
+	  {
+	    vst2_mgr_get_dir_plugins(vst2_mgr, file_name);
+	  }
+        else {
+            char *ext = strrchr(file_name, '.');
+            if (ext
+                && (strcmp(ext, ".so") == 0 || strcmp(ext, ".dll") == 0
+                    || strcmp(ext, ".dylib") == 0)) {
```

Comment:
According to my SwankyAmp download this may need to include `.vst3` which suggests that `.vst` and `.vst2` might also be checked.

Also, these need to be case-insensitive comparisons. Maybe LADSPA and LV2 can be case sensitive given their Linux heritage, but definitely given VST's Windows heritage it should be case insensitive.

According to [this Ardour discussion](https://discourse.ardour.org/t/how-to-properly-load-vst3-on-linux/105022), a `.vst3` is a bundle (folder), but SwankyAmp gave me a DLL:
```
file '/home/ddennedy/Downloads/SwankyAmp.vst3' 
/home/ddennedy/Downloads/SwankyAmp.vst3: PE32+ executable (DLL) (GUI) x86-64, for MS Windows
```
Between natively compiled VST plugins and those that can run via WINE, I see that there is going to be a lot of confusion ahead.

Reply:
For this, please simply do a `strcasecmp()` for the DLL.

Reply:
yeah there is `.vst` and for vst3 there is `.vst3`

https://helpcenter.steinberg.de/hc/en-us/articles/115000177084-VST-plug-in-locations-on-Windows
https://helpcenter.steinberg.de/hc/en-us/articles/115000171310-VST-plug-in-locations-on-Mac-OS-X-and-macOS